src/rng.cpp

```diff
@@ -5,35 +5,49 @@
 #include <algorithm>
 #include <utility>
 
-long rng( long val1, long val2 )
+#include "cata_utility.h"
+
+static std::uniform_int_distribution<int> rng_int_dist;
+static std::uniform_real_distribution<double> rng_real_dist;
+static std::normal_distribution<double> rng_normal_dist;
+static std::uniform_int_distribution<unsigned int>
+rng_uint_dist( 0, std::numeric_limits<unsigned int>::max() );
+
+unsigned int rng_bits()
 {
-    long minVal = ( val1 < val2 ) ? val1 : val2;
-    long maxVal = ( val1 < val2 ) ? val2 : val1;
-    return minVal + long( ( maxVal - minVal + 1 ) * double( rand() / double( RAND_MAX + 1.0 ) ) );
+    return rng_uint_dist( rng_get_engine() );
 }
 
-double rng_float( double val1, double val2 )
+int rng( int lo, int hi )
 {
-    double minVal = ( val1 < val2 ) ? val1 : val2;
-    double maxVal = ( val1 < val2 ) ? val2 : val1;
-    return minVal + ( maxVal - minVal ) * static_cast<double>( rand() ) / static_cast<double>
-           ( RAND_MAX + 1.0 );
+    if( lo > hi ) {
+        std::swap( lo, hi );
+    }
+
+    decltype( rng_int_dist.param() ) range( lo, hi );
+    rng_int_dist.param( range );
+    return rng_int_dist( rng_get_engine() );
 }
 
-bool one_in( int chance )
+double rng_float( double lo, double hi )
 {
-    return ( chance <= 1 || rng( 0, chance - 1 ) == 0 );
+    if( lo > hi ) {
+        std::swap( lo, hi );
+    }
+
+    decltype( rng_real_dist.param() ) range( lo, hi );
+    rng_real_dist.param( range );
+    return rng_real_dist( rng_get_engine() );
 }
 
-//this works just like one_in, but it accepts doubles as input to calculate chances like "1 in 350,52"
-bool one_in_improved( double chance )
+bool one_in( int chance )
 {
-    return ( chance <= 1 || rng_float( 0, chance ) < 1 );
+    return ( chance <= 1 || rng( 0, chance - 1 ) == 0 );
 }
 
 bool x_in_y( double x, double y )
 {
-    return ( static_cast<double>( rand() ) / RAND_MAX ) <= ( static_cast<double>( x ) / y );
+    return rng_float( 0, 1 ) * y <= x;
```

Comment:
`rng_float( 0, y )`?  (Unless that doesn't work for `y == 0` or something)

Reply:
Sure thing.